neuralpp/symbolic/expression.py

```diff
@@ -117,8 +117,33 @@ def contains(self, target: Expression) -> bool:
 
     @abstractmethod
     def syntactic_eq(self, other) -> bool:
+        """
+        Returns if self and other are syntactically equal, i.e.,
+        that they are of subclass of Expression and their internal representation are equal.
+        This method usually depends on subclass-specific library calls,
+        e.g., Z3Expression.syntactic_eq() would leverage z3.eq().
+        This method should be considered as a cheap way to check syntactic equality of two symbolic expressions.
+        """
         pass
 
+    def structure_eq(self, other) -> bool:
+        """
+        Returns if self and other are "structurally" equivalent, i.e., that they have the same Expression interfaces.
+        E.g, a Z3Expression of "a + b" does not syntactic_eq() a SymPyExpression of "a + b", but a call of
+        structure_eq() on the two should return True.
+        This method is general and more expensive than syntactic_eq()
+        """
```

Comment:
Introduced a new `structure_eq` method to help writing tests. Don't know how the name sounds, I'm also thinking of `interface_eq`, suggestions?

Reply:
`structure_eq` sounds quite reasonable to me :)